### ISSUE TITLE
[Bug] Can't install Chia Blochchain GUI on Ubuntu 20.04

### ISSUE THREAD

**Issue by SecurID** [Chia-Network/chia-blockchain#9542]
### What happened?

Tried to install chia including gui on a fresh ubuntu 20.04.

I read in another issue that it is related to a moved dependency. but main branch doesn't work too. 
Anyone knows which is the latest functioning branch? 
Or any other help?

### Version

latest

### What platform are you using?

Linux

### What ui mode are you using?

GUI

### Relevant log output

```shell
npm ERR! code 1
npm ERR! Command failed: git checkout master
npm ERR! error: Pfadspezifikation 'master' stimmt mit keinen git-bekannten Dateien überein
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kupferschmidt/.npm/_logs/2021-12-10T23_11_55_020Z-debug.log
```

**Comment by backyardengine** [comment]
I have also had this issue the past few days.  Tried on Ubuntu and MintOS on different computers.

**Comment by shaneo257** [comment]
try using the installer from https://www.chia.net/download/

**Comment by FedericoLaTorre** [comment]
I have this issue when trying to install GUI

The authenticity of host 'github.com (20.201.28.151)' can't be established.et
ECDSA key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
[              ....] - install dependencies: verb installed leaf @chia/wallet

It remains there forever.
I removed the key just in case.

**Comment by number435398** [comment]
Lets just face facts; the installer for Chia has never worked in Ubuntu.

**Comment by number435398** [comment]
Also, I've had the same issue. After spending hours trying to fix it what I found was it seems to be sourced in the following: When going to `sh install-gui.sh` (and perhaps even in the previous line of `sh install.sh`) NPM is unable to extract all the files because it says:

`"npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!". `

Basically, it looks like someone updated the files to be used with that new version and npm can't handle it. But you will still follow the install instructions and when you go to the final command to execute the program, it doesn't work because that earlier process in the installation failed.

**Comment by shaneo257** [comment]
> 

i have used the installer downloaded from https://www.chia.net/download/ for ubuntu a number of times and it works fine

**Comment by number435398** [comment]
Every installer I've tried for Chia in Ubuntu has never worked. So I went through the unnecessarily laborious process of torrenting the specific one you mentioned. I've run it like 6 times. It never installs anything.

**Comment by emlowe** [comment]
What command did you use to install it? typically I do `sudo dpkg -i package_file.deb` and also, it puts files in a somewhat odd place. It places chia-blockchain executable link in /usr/bin/chia-blockchain and then most files go into /usr/lib/chia-blockchain/

**Comment by emlowe** [comment]
You can use `sudo dpkg --debug=10 -i file.deb` and it should output all the files it processes as well as various other details

**Comment by emlowe** [comment]
You can also just direct download the file `https://download.chia.net/latest/x86_64-Ubuntu-gui` should download the file directly to you.

**Comment by number435398** [comment]
Wow, emlowe tried to provide some help instead of just silencing someone and closing out their thread. Color me surprised.

Also, your attempt was not conducive to anything productive, it didn't work. It'd be nice if the original problem in this thread was just, you know, fixed. I've spent days now just trying to open a cold wallet and can't due to this issue. I mean, like, wow, seriously.

**Comment by emlowe** [comment]
Can you post the output of `sudo dpkg --debug=10 -i file.deb` please? Does it throw any errors? I need a little help here because I have no idea what is going wrong. "didn't work" isn't informative enough for me to help you.

I'll note, the original issue that was made here is due to problems introduced by 3rd party repos after 1.2.11 was tagged, so using `git clone -b latest` will continue to have problems until we release a new version that updates "latest" to point to that. To the end , yes it remains a problem to install using git. However, we are not aware of any problems with using the DEB file, so please provide some more information if you can

**Comment by FedericoLaTorre** [comment]
Still stuck at installing.

lerna notice cli v4.0.0
lerna info Bootstrapping 7 packages
lerna info Installing external dependencies
The authenticity of host 'github.com (20.201.28.151)' can't be established.rency 4
ECDSA key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
[              ....] | install dependencies: verb installed leaf @chia/wallet

**Comment by emlowe** [comment]
This issue is about installing `latest` or `1.2.11`. If you have issues with `main`, it would be best to use a different GH issue with your details.

However, in this case, I again recommend using the beta packages we make available under Latest Beta on https://www.chia.net/download/

Note this particular issue was caused by NPM changing to using SSH instead of HTTPS for fetching items from github. This is an outstanding issue in main that will be resolved soon.

**Comment by number435398** [comment]
Your instructions for the use of the deb file don't appear to be complete from download to opening the program. I recall no errors.
So when is the normal way of doing it going to be fixed?

**Comment by shaneo257** [comment]
the preferred way for new installs/upgrades is to use the installers from https://www.chia.net/download/

**Comment by number435398** [comment]
I guess its different for every project. I've always understood that gitting and compiling it yourself was the preferred way. well, I'm not against new things, but I have found no step by step instructions for the installer way of doing it either. whenever i've tried an installer for chia in ubuntu I always get "This file is not supported" or something like that from the resident program installer. I didn't know it required actual terminal commands.

**Comment by emlowe** [comment]
You don't have to use the terminal, but I find it's the simplest way. We will update our documentation.

For the record - there are multiple ways to install DEB packages on Ubuntu: https://help.ubuntu.com/kubuntu/desktopguide/C/manual-install.html

**Comment by number435398** [comment]
I right clicked the deb file and clicked it to be opened by Software Install (the only available option on a fresh Ubuntu install) as per that link's recommendations. The result was it appeared to begin processing it/opening it and then, as far as I can tell, did nothing.

**Comment by backyardengine** [comment]
Search yours programs, when its installed it will look like nothing happened.  If not use dpkg to install, very easy.

**Comment by ChiaMineJP** [comment]
Hi all.
Please try to upgrade npm version to 8, if you have npm v6 or lower installed.

**Comment by ChiaMineJP** [comment]
FYI
`install-gui.sh` and `install.sh` was overhauled in the PR below.
https://github.com/Chia-Network/chia-blockchain/pull/9921

You will have working installers in the next release.
Thank you for your patience.
